ament_cmake_core/cmake/index/ament_index_get_resources.cmake

```diff
@@ -52,10 +52,17 @@ function(ament_index_get_resources var resource_type)
   list(INSERT paths_to_search 0 "${CMAKE_INSTALL_PREFIX}")
   set(all_resources "")
   foreach(path IN LISTS paths_to_search)
+    set(resource_index_path "${path}/share/ament_index/resource_index")
     file(GLOB resources
-      RELATIVE "${path}/share/ament_index/resource_index/${resource_type}"
-      "${path}/share/ament_index/resource_index/${resource_type}/*")
-    list_append_unique(all_resources ${resources})
+      RELATIVE "${resource_index_path}/${resource_type}"
+      "${resource_index_path}/${resource_type}/*")
+    foreach(resource IN LISTS resources)
+      # Ignore all subdirectories, and any files starting with a dot
+      if((NOT EXISTS "${resource_index_path}/${resource_type}/${resource}/") 
+          AND (NOT resource MATCHES "[.].+"))
```

Comment:
Since the comment suggests to match all entries starting with a dot (which includes just a dot) the regex should be `[.].*`. But in general I would avoid a regex for a trivial check like this. Instead a string comparison is sufficient and more efficient:

    string(SUBSTRING "${resource}" 0 1 resource_char0)
    if("${resource_char0} " STREQUAL ". ")